test-suite/src/foreign_key.rs

```diff
@@ -87,42 +87,422 @@ test_case!(foreign_key, {
     )
     .await;
 
-    g.named_test(
-        "Unsupported foreign key option: CASCADE",
-        "CREATE TABLE ReferencingTable (
+    // We create a table with a foreign key constraint that references another table
+    // which includes a ON DELETE CASCADE clause.
+    g.run(
+        "CREATE TABLE ReferencingTableCascade (
             id INT,
             name TEXT,
             referenced_id INT,
-            FOREIGN KEY (referenced_id) REFERENCES ReferencedTableWithPK (id) ON DELETE CASCADE
+            FOREIGN KEY (referenced_id) REFERENCES ReferencedTableWithPK (id) ON UPDATE CASCADE ON DELETE CASCADE
         );",
-        Err(TranslateError::UnsupportedConstraint("CASCADE".to_owned()).into()),
     )
     .await;
 
+    // We insert a row into the referenced table.
+    g.run("INSERT INTO ReferencedTableWithPK VALUES (1, 'referenced_table1'), (2, 'referenced_table2');")
+        .await;
+
+    // We insert a row into the referencing table.
+    g.run("INSERT INTO ReferencingTableCascade VALUES (1, 'referencing_table with referenced_table', 1), (2, 'referencing_table with referenced_table', 2), (3, 'referencing_table with referenced_table', 1);")
+        .await;
+
+    // We update the row in the referenced table, which should yield a cascade update
+    // with the one row in the referencing table being updated, and the other rows from
+    // the referencing table being deleted.
     g.named_test(
-        "Unsupported foreign key option: SET DEFAULT",
-        "CREATE TABLE ReferencingTable (
+        "Updating referenced row should update referencing row as well",
+        "UPDATE ReferencedTableWithPK SET name = 'referenced_table1 updated' WHERE id = 1;",
+        Ok(Payload::Update(1)),
+    )
+    .await;
+
+    // We check that the row from the referencing table has been deleted as well.
+    g.named_test(
+        "Referencing row should be deleted on update cascade",
+        "SELECT * FROM ReferencingTableCascade WHERE id = 1;",
+        Ok(Payload::Select {
+            labels: vec![
+                "id".to_owned(),
+                "name".to_owned(),
+                "referenced_id".to_owned(),
+            ],
+            rows: vec![],
+        }),
+    )
+    .await;
```

Comment:
These look wrong tests.
`ON UPDATE CASCADE` means If the refereced key is updated, referencing value should be updated cacadely.

Tests should be like this
```rs
    g.named_test(
        "Updating referenced row should update referencing row as well",
        "UPDATE ReferencedTableWithPK SET id = 3 WHERE id = 1;",
        Ok(Payload::Update(1)),
    )
    .await;

    g.named_test(
        "Referencing row should be deleted on update cascade",
        "SELECT id FROM ReferencingTableCascade WHERE referenced_id = 3;",
        Ok(Payload::Select {
            labels: vec!["id".to_owned()],
            rows: vec![vec![Value::I64(1), Value::I64(3)]],
        }),
    )
    .await;
```

But currently you will get `UpdateOnPrimaryKeyNotSupported`. Cuz we decided to prohibit updating PK.

So i suggest,
1. Implement FK on UNIQUE
2. Write test cases of `ON UPDATE CASCADE` of `ReferencedTableWithUnique`
3. Fix `ON UPDATE CASCADE` logic from deleting to updating

Reply:
If `UpdateOnPrimaryKeyNotSupported`, then `ON UPDATE CASCADE` should be just illegal, right?

Reply:
Ok so, I have now removed the wrong implementation of the on update X in the update method and rewrote the test suite regarding the use of the ON UPDATE x. Now the tests are correct. I have separated in two documents the tests regarding the ON UPDATE and the ON DELETE, since the expected behaviour is very different from what a fully working implementation should be. I have documented on the top of the ON UPDATE test suite the reason for this.

This was one in commit [b962010](https://github.com/gluesql/gluesql/pull/1517/commits/b96201060a34ffe9c16c98f2f5187b6908bae04e)